server/Arcadia.Assistant.CSP/CspDepartmentsStorage.cs

```diff
@@ -17,12 +18,28 @@ public class CspDepartmentsStorage : DepartmentsStorage
 
         private readonly CspConfiguration configuration;
 
+        private string lastErrorMessage;
+
         public CspDepartmentsStorage(Func<ArcadiaCspContext> contextFactory, CspConfiguration configuration)
         {
             this.contextFactory = contextFactory;
             this.configuration = configuration;
         }
 
+        protected override void OnReceive(object message)
+        {
+            switch (message)
+            {
+                case GetHealthCheckStatusMessage _:
```

Comment:
maybe put this code in `DepartmentsStorage`? So you won't have a) OnRecieve, b) huge try/catch - just use onFailure callback in `PipeTo` function?

Reply:
oh, nevermind b) part. I think you need to pipe it to self in order to make it work

Reply:
Then we obligate dependents of `DepartmentsStorage` to implement this logic even if it uses another health check procedure.